### ISSUE TITLE
help request: i want konw plugin can work before match route

### ISSUE THREAD

**Issue by nulinuli-nuli** [apache/apisix#10676]
### Description

i want build a plugin before match route，can it work?

### Environment

- APISIX version (run `apisix version`):
- Operating system (run `uname -a`):
- OpenResty / Nginx version (run `openresty -V` or `nginx -V`):
- etcd version, if relevant (run `curl http://127.0.0.1:9090/v1/server_info`):
- APISIX Dashboard version, if relevant:
- Plugin runner version, for issues related to plugin runners:
- LuaRocks version, for installation issues (run `luarocks --version`):

**Comment by kayx23** [comment]
Could you rephrase your question with bit more details? Don't understand what's being asked.

**Comment by nulinuli-nuli** [comment]
i want add a special head to request before matchroute

**Comment by nulinuli-nuli** [comment]
![image](https://github.com/apache/apisix/assets/55182372/bd7a82e9-3f75-4cd5-9cde-7c4c50c8d8ef)

**Comment by kayx23** [comment]
You would like to build a global plugin?

**Comment by shreemaan-abhishek** [comment]
ping @nulinuli-nuli

**Comment by nulinuli-nuli** [comment]
> ping @nulinuli-nuli

Which version can be supported and how long will it take?

**Comment by shreemaan-abhishek** [comment]
we don't understand your problem

**Comment by Vacant2333** [comment]
@nulinuli-nuli under my understanding u want a phase like `rewrite` 
<img width="345" alt="image" src="https://github.com/apache/apisix/assets/19872346/ad568c76-9b95-4445-a127-d0de8b7871cc">
and the phase u want `before_match_route` phase to make some actions?

**Comment by nulinuli-nuli** [comment]
> @nulinuli-nuli under my understanding u want a phase like `rewrite` <img alt="image" width="345" src="https://private-user-images.githubusercontent.com/19872346/298001787-ad568c76-9b95-4445-a127-d0de8b7871cc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ1NzIzNTcsIm5iZiI6MTcyNDU3MjA1NywicGF0aCI6Ii8xOTg3MjM0Ni8yOTgwMDE3ODctYWQ1NjhjNzYtOWI5NS00NDQ1LWExMjctZDBkZThiNzg3MWNjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI1VDA3NDczN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRlYTE4MWMyZmQzZmVlMWUxNzVlMzk1NDBiMzRkMGNiMTI0OGY4ZDIyOGRlZTY1MGFiOGEwNjhmNmZlN2Q3MDcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.LG-PTEZIufFTyWCH4IKAEWIAV1ZI2FDYZ8Slpq1RJTE"> and the phase u want `before_match_route` phase to make some actions?

yes

**Comment by nulinuli-nuli** [comment]
i want  add http head before matching route，use java build plugin
